rfcs/0130-stalled-rfcs.md

```diff
@@ -0,0 +1,145 @@
+---
+feature: stalled-rfcs
+start-date: 2022-06-01
+author: kevincox
+co-authors: spacekookie
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+This defines two alternate "end states" for RFCs.
+
+The "Awaiting Author" state defines an RFC that isn't ready to be pushed forward and is intentionally stalled. The NixOS RFC Steering Committee will not ping or push this RFC until it returns to the main workflow. These RFCs will be put into a draft status until they are ready to re-enter the process.
+
+The "Lacking Interest" state defines an RFC that can not gather a shepherd team. This is taken to indicate that there is not enough interest in the Nix community. These RFCs will be Closed until more interest is gathered.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS RFC process describes a process with a focus on moving RFCs toward completion. This is where an engaged author and community push an RFC to a decision, either to accept or reject the proposal.
+
+However things don't always work like this. Sometimes the author doesn't have sufficient time or motivation to keep iterating on the RFC and sometimes the community doesn't have enough interested members to staff a shepherd team.
+
+It is important that these "alternate exits" are well defined to avoid wasting effort trying to push a stalled RFC forward and to set clear expectations for all participants.
+
+TL;DR timeouts are critical in distributed systems.
+
+# Detailed design
+[design]: #detailed-design
+
+Here is an updated diagram for the RFC process.
+
+```mermaid
```

Comment:
The rendered graph looks incorrect; "On Hold" also goes through "Author Withdraws" and ends at "Closed - Withdrawn", but both of those sound wrong to me. Shouldn't there be a separate end state for "On Hold"?

Reply:
Hmm, you are right but I don't know how to fix this. Suggestions welcome.

Reply:
cc @ryantm Let's keep the diagram discussion here.

I'm not sure if mermaid supports manual positioning so I see a few options:

1. Ignore the rendering bug.
2. Change `mermaid` to `text` so that it doesn't render and the source is the visible thing.
3. Recreate the diagram with another tool. (Harder to edit unfortunately).
4. Drop the diagram completely.
5. Block this on a mermaid rendering bug.

Personally I am in favour of 1 but if people feel strongly I could do 2 or 3.

Reply:
It's just awkward rendering -- the "On Hold" label corresponds to the arrow to its left (from "Discuss and Refine" to "Draft"). See my suggestion for a hacky fix.

Reply:
Thanks, that's a clever workaround for lack of manual positioning 🙂